mongodbatlas/resource_mongodbatlas_cluster.go

```diff
@@ -1175,7 +1178,16 @@ func expandReplicationSpecs(d *schema.ResourceData) ([]matlas.ReplicationSpec, e
 					id = oldID
 				}
 			}
-			regionsConfig, err := expandRegionsConfig(spec["regions_config"].(*schema.Set).List())
+
```

Comment:
I just have one doubt in the logic, this is suppose to work with just one region, but of course it works, my only concern is that the change detection is inside the expand , of course it's called by the Update. but no sides effects when this is more than one region right?

Reply:
@coderGo93 did this concern get addressed?

Reply:
Hi, I tried to add another region and it didn't work because the configuration was not correct from API error response, but when I tested, it worked in request but somehow after it's over in response for replication specs didn't change but it worked for region_name and with re plan it appears not changes.

Reply:
@coderGo93  not sure I'm following - so when you tested more than one region it worked but the API still errored?

Reply:
@themantissa Not because it seems it's not possible to do that, the error is like this `400 (request "INVALID_CLUSTER_CONFIGURATION") The specified cluster configuration is not valid.` or Unexpected error , I guess that happens when you set region name along with replication specs and cluster type, but it works without appearing changes after re plan with the config example from the ticket INTMDB-200 without needing to use the replication specs schema

Reply:
Hi @coderGo93 

Let's try to clarify all doubts.

**The use case**

if someone creates with only one region, on behalf of the user we populate the replication_specs with just one region.

The issue 

If the customer just updates the region name, we didn't update the only region in their replication specs. So that was the fix. 

The concerns. 

The fix works, but take in mind we do have different type of clusters , from the documentation https://docs.atlas.mongodb.com/reference/api/clusters-create-one/, and the code fix is inside the expand function so it does not validate the cluster type. So let's say someone by mistake starts playing in GCP with the provider region name, and you detect that change in the expand function you will overwrite the zones for all the items. 

So before merging this two options move your fix outside the expand function or in the other hand, just validate that the API actually complains :) . otherwise we will end up with suprises.

Reply:
Pushed changes by adding validation when cluster type is REPLICASET along if provider is GCP in update. Also I added to check if the original region is equal to one element of replication specs so it will replace one element instead of all regions.
Steps:
1. Create a cluster with `provider_region_name`
2. After created, update the `provider_region_name`
3. Wait for update to be finished, it can take a while from 10 min to 1hr
4. After updated, with terraform plan shows no changes infrastructure

Config file used for testing
```hcl
resource "mongodbatlas_cluster" "terraform_cluster3" {
  project_id = var.project_id
  name       = "test-cluster1"
  auto_scaling_disk_gb_enabled = true
  mongo_db_major_version       = "4.4"
  provider_name               = "GCP"
  disk_size_gb                = 10
  provider_instance_size_name = "M10"
  num_shards = 1
  provider_region_name        = "US_WEST_2"
}

```

Reply:
@coderGo93  Could you also test that everything is working for clusterType `SHARDED` and `GEOSHARDED` with more than 1 shard, please?

Reply:
Yes, I just testd and it worked for me, it only will occur when the provider is GCP and  cluster type is REPLICASET with additional code when it should replace one element instead of all elements in update.

Reply:
Thank you very much!